engine/core/src/main/java/org/pentaho/reporting/engine/classic/core/layout/output/AbstractOutputProcessor.java

```diff
@@ -63,7 +98,11 @@ public final LogicalPageKey getLogicalPage( final int page ) {
       throw new IllegalStateException();
     }
 
-    return logicalPages.get( page );
+    try {
+      return logicalPages.get( page );
+    } catch ( IndexOutOfBoundsException e ) {
+      throw e;
+    }
```

Comment:
As there's no extra handling when an IndexOutOfBoundsException is caught, both old and new code behaves the same way: any exception thrown by `logicalPages.get( page )` will always be thrown back!
As so, the try-catch is dispensable and it's best to leave the code as it was.